tests/kafkatest/services/verifiable_consumer.py

```diff
@@ -275,7 +275,7 @@ def stop_node(self, node, clean_shutdown=True):
         # block until the corresponding thread exits
         if len(self.worker_threads) >= self.idx(node):
             # Need to guard this because stop is preemptively called before the worker threads are added and started
-            self.worker_threads[self.idx(node) - 1].join()
```

Comment:
I think you need to do something like `assert not self.worker_threads[self.idx(node)-1].isAlive()` after this since `join()` always returns `None`.